Add parameter hold_secs for Harmony remote send command

ehendrix23 opened the pull request:
## Description:
Add optional parameter to specify how long the remote should "hold" a pressed button before releasing it.

Architecture issue home-assistant/architecture#90 was originally opened for adding this parameter as well.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#7985

## Example entry for `configuration.yaml` (if applicable):
```yaml
- platform: harmony
  name: Living Room 
  activity: Watch TV
  hold_secs: 1
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

ehendrix23 committed:
Update requirements

Updated requirements

ehendrix23 committed:
Add attributes

Add firmware and config version attributes

ehendrix23 committed:
Add hold for button press on send_command

ehendrix23 committed:
Small bump for aioharmony

Small version bump increase for aioharmony

ehendrix23 committed:
Fix requirements file

For some reason aioharmony ended up in there as a duplicate. Fixed it.

ehendrix23 committed:
Revert rebase changes

Revert some changes that should have been reverted back as part of rebase.

amelchio commented on homeassistant/components/remote/harmony.py:
```diff
@@ -347,6 +349,9 @@ def new_activity(self, activity_info: tuple) -> None:
         delay_secs = kwargs.get(ATTR_DELAY_SECS, self._delay_secs)
+        hold_secs = kwargs.get(ATTR_HOLD_SECS)
+        _LOGGER.debug("Sending commands to device %s holding for %s seconds",
+                      device, hold_secs)
```
Why not log `delay_secs` as well?

ehendrix23 replied:
Cause I do not use it? :-)

Makes perfect sense, will add it.

amelchio commented on homeassistant/components/remote/harmony.py:
```diff
@@ -41,6 +41,8 @@
     vol.Optional(ATTR_DELAY_SECS, default=DEFAULT_DELAY_SECS):
         vol.Coerce(float),
+    vol.Optional(ATTR_HOLD_SECS, default=DEFAULT_HOLD_SECS):
+        vol.Coerce(float),
```
You extend the platform schema but I don't see that you ever read the configuration?

I am actually not sure that it makes sense to have a platform default for `hold_secs`, wouldn't it likely be different for each command?

balloob replied:
I don't understand why these are even part of the config, the config should just set up the connection and that's it ? 

Let's not add this one, in a future PR remove the other ones.

ehendrix23 replied:
Makes sense, removed it. I'll open a future PR to remove the other one (after I'm done with XMPP for aioharmony, just wanted to take a quick break from that one).

amelchio commented on homeassistant/components/remote/harmony.py:
```diff
@@ -347,6 +349,9 @@ def new_activity(self, activity_info: tuple) -> None:
 
         num_repeats = kwargs.get(ATTR_NUM_REPEATS)
         delay_secs = kwargs.get(ATTR_DELAY_SECS, self._delay_secs)
+        hold_secs = kwargs.get(ATTR_HOLD_SECS)
```
Since `ATTR_HOLD_SECS` has a default, use `kwargs[ATTR_HOLD_SECS]`.

(Yes, I know it's done wrong just two lines above. Whoever wrote that should ... oh – it was me ... well, times were different back then and I was young and reckless)

ehendrix23 replied:
LOL, does this mean I'm young and reckless as well? I like that <g>.
Changed.

ehendrix23 committed:
Updated based on review

Removed HOLD_SECS from platform schema (configuration)
Updated getting kwargs in async_send_command
Updated debug log to include delay_secs

amelchio reviewed (approved):
Looks good!
